code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -376,7 +376,11 @@
 /mob/living/silicon/robot/proc/self_destruct(mob/usr)
 	var/turf/groundzero = get_turf(src)
 	message_admins(span_notice("[ADMIN_LOOKUPFLW(usr)] detonated [key_name_admin(src, client)] at [ADMIN_VERBOSEJMP(groundzero)]!"))
-	log_game("[key_name(usr)] detonated [key_name(src)]!")
+	usr.log_message("detonated [key_name(src)]!", LOG_GAME)
+	usr.log_message("detonated [key_name(src)]!", LOG_ATTACK, log_globally = FALSE)
+	src.log_message("was detonated by [key_name(usr)]!", LOG_VICTIM)
+	src.log_message("was detonated by [key_name(usr)]!", LOG_ATTACK, log_globally = FALSE)
```

Comment:
Does this need to be logged twice for the victim?

Reply:
i suppose not